nmdc_server/ingest/envo.py

```diff
@@ -165,6 +165,32 @@ def _nested_envo_subtree(
     ]
 
 
+def _prune_useless_nodes(
+    node: EnvoTreeNode,
+    present_terms: Set[str],
+    parent: Optional[EnvoTreeNode] = None,
+    modified: bool = False,
+) -> bool:
+    """
+    Remove useless internal (non-root) nodes from a tree.
```

Comment:
Could you provide a definition of "useless" in this docstring?

Reply:
I'll give my best attempt, but I don't want to sound too authoritative with our working definition because I just don't feel terribly confident in the underlying math.